Add stats agg api (#524)
timescale/docs: api/stats_aggs.md

Some hyperfunctions are included in the default TimescaleDB product. For
additional hyperfunctions, you need to install the

> [Timescale Toolkit][install-toolkit] PostgreSQL extension, all of these aggregates 
> are in the [Timescale Toolkit][install-toolkit]. They provide a superset of functionality
> available with [PostgreSQL statistical aggregates][pg-stats-aggs

Reviewer: ```suggestion
[Timescale Toolkit][install-toolkit] PostgreSQL extension. All of these aggregates 
are in the [Timescale Toolkit][install-toolkit]. They provide a superset of functionality
available with [PostgreSQL statistical aggregates][pg-stats-aggs].
```

> Note that statistical aggregates have 1D aggregates which are computed for a single variable
> and 2D or regression aggregates which are computed over two variables. All 1D aggregates are available
> for each of the 2D variables separately, this is denoted by `_y` or `_x` following the name. 
> For more information about statistical aggregate functions, see the
> [hyperfunctions documentation][hyperfunctions-stats-aggs].

Reviewer: ```suggestion
Statistical aggregates have one-dimensional (1D) aggregates which are 
computed for a single variable, and two-dimensional (2D) or regression 
aggregates which are computed over two variables. All 1D aggregates are 
available for each of the 2D variables separately, this is denoted by `_y` 
or `_x` following the name. 

For more information about statistical aggregate functions, see the
[hyperfunctions documentation][hyperfunctions-stats-aggs].
```

> 

Reviewer: ```suggestion
```

|Hyperfunction family|Types|API Calls|Included by default|Toolkit required|
|-|-|-|-|-|

> |Statsistical Aggregates|Statistical Aggregates|[`stats_agg`](/hyperfunctions/stats_aggs/stats_agg/)|❌|✅|

Reviewer: ```suggestion
|Statistical Aggregates|Statistical Aggregates|[`stats_agg`](/hyperfunctions/stats_aggs/stats_agg/)|❌|✅|
```
